packages/salesforcedx-vscode-apex/package.json

```
@@ -226,6 +226,11 @@
           "default": null,
           "description": "%java_home_description%"
         },
+        "salesforcedx-vscode-apex.java.memory": {
+          "type": "integer",
+          "default": 4096,
```

This default is a bit high if we consider the minimum system requirements for running VSCode https://code.visualstudio.com/docs/supporting/requirements. Based on what we've seen on our testing, small to medium projects do well with 1 or 2 GB. It's on larger projects where a higher value would help. @sfsholden @rcoringrato-sfdc what do you think ?

cc @ntotten

---

@lcampos I agree and am happy to update the PR. Would the following be a resolution:

```suggestion
          "default": 1024,
```

and a similar change at line 37 of requirements.ts?

---

Sorry @kenhuman I take back my previous comment. After spending more time testing and going through the Java documentation I think the change should be to exclusively override the JVM defaults if the setting is set and not for us to provide a hardcoded default. This is because the JVM allocates 1/4th of your physical memory. For example, in a machine that has 8GB of RAM the JVM is already allocating 2 for this process.

The purpose of this change is to help users running into heap size issues override the default values and allow those users that are not running into those issues still use the JVM defaults. Let me know what you think.

---

@lcampos I have updated this PR. The java.memory configuration variable now defaults to null, causing the JVM to use its default value when allocating memory.